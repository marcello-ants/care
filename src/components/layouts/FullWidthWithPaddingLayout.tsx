import { Container, makeStyles } from '@material-ui/core';
import { ILayoutProps } from '@/types/layout';

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    overflowX: 'hidden',
    flex: '1 0 auto',
    [theme.breakpoints.up('md')]: {
      paddingTop: theme.spacing(6),
    },
  },
  content: {
    margin: 'auto',
    maxWidth: '410px',
    [theme.breakpoints.up('md')]: {
      paddingLeft: '80px',
      paddingRight: '80px',
      maxWidth: '100%',
    },
  },
}));

export default function FullWidthWithPadding({ children }: ILayoutProps) {
  const classes = useStyles();
  return (
    <main>
      <Container maxWidth="lg" className={classes.root}>
        <div className={classes.content}>{children}</div>
      </Container>
    </main>
  );
}
