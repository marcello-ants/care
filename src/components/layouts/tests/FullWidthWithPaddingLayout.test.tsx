import { render } from '@testing-library/react';
import FullWidthWithPaddingLayout from '../FullWidthWithPaddingLayout';

describe('FullWidthWithPadding Layout component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<FullWidthWithPaddingLayout>test</FullWidthWithPaddingLayout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
