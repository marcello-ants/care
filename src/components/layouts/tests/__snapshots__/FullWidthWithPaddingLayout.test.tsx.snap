// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullWidthWithPadding Layout component matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-content-2"
      >
        test
      </div>
    </div>
  </main>
</DocumentFragment>
`;
